File: bindings/lua/Polycode/EntityProp.lua
class "EntityProp"


function EntityProp:__getvar(name)
	if name == "propName" then
		return Polycode.EntityProp_get_propName(self.__ptr)
	elseif name == "propValue" then
		return Polycode.EntityProp_get_propValue(self.__ptr)
	end
end

function EntityProp:__setvar(name,value)
	if name == "propName" then
		Polycode.EntityProp_set_propName(self.__ptr, value)
		return true
	elseif name == "propValue" then
		Polycode.EntityProp_set_propValue(self.__ptr, value)
		return true
	end
	return false
end
function EntityProp:__delete()
	if self then Polycode.delete_EntityProp(self.__ptr) end
end

File: bindings/javascript/Polycode/EntityProp.js
function EntityProp() {
	Object.defineProperties(this, {
		'propName': { enumerable: true, configurable: true, get: EntityProp.prototype.__get_propName, set: EntityProp.prototype.__set_propName},
		'propValue': { enumerable: true, configurable: true, get: EntityProp.prototype.__get_propValue, set: EntityProp.prototype.__set_propValue}
	})
}
EntityProp.prototype.__get_propName = function() {
	return Polycode.EntityProp__get_propName(this.__ptr)
}

EntityProp.prototype.__set_propName = function(val) {
	Polycode.EntityProp__set_propName(this.__ptr, val)
}

EntityProp.prototype.__get_propValue = function() {
	return Polycode.EntityProp__get_propValue(this.__ptr)
}

EntityProp.prototype.__set_propValue = function(val) {
	Polycode.EntityProp__set_propValue(this.__ptr, val)
}

